enhancements/installer/cluster-api-based-installations.md

```diff
@@ -0,0 +1,485 @@
+
+1 directory, 12 files
+```
+
+The manifests within this `cluster-api` directory will not be written to the cluster or included in bootstrap ignition.
+In future work, we expect these manifests to be pivoted to the cluster to enable the target cluster to take over managing
+its own infrastructure.
+
+The Cluster API manifests will be generated with `.spec` fields in the [manifest asset target][asset] with `.status`
+updated after cluster creation.
+
+#### Infrastructure Provisioning
+
+##### CAPI Provisioning
+
+The CAPI infrastructure manifests determine the infrastructure resources that will be provisioned by the CAPI controllers.
+Here is an example AWS infra cluster manifest from the proof-of-concept implementation:
+
+```shell
+$ cat cluster-api/02_infra-cluster.yaml
+apiVersion: v1beta2
+kind: AWSCluster
+metadata:
+  creationTimestamp: null
+  name: capi-example-kbdpg
+  namespace: openshift-cluster-api-guests
+spec:
+  bastion:
+    enabled: false
+  controlPlaneEndpoint:
+    host: ""
+    port: 0
+  controlPlaneLoadBalancer:
+    additionalListeners:
+    - port: 22623
+      protocol: TCP
+    crossZoneLoadBalancing: false
+    loadBalancerType: nlb
+    name: capi-example-kbdpg-ext
+    scheme: internet-facing
+  network:
+    additionalControlPlaneIngressRules:
+    - description: MCS traffic from cluster network
+      fromPort: 22623
+      protocol: tcp
+      sourceSecurityGroupRoles:
+      - node
+      - controlplane
+      toPort: 22623
+    - description: controller-manager
+      fromPort: 10257
+      protocol: tcp
+      sourceSecurityGroupRoles:
+      - controlplane
+      - node
+      toPort: 10257
+    - description: kube-scheduler
+      fromPort: 10259
+      protocol: tcp
+      sourceSecurityGroupRoles:
+      - controlplane
+      - node
+      toPort: 10259
+    cni:
+      cniIngressRules:
+      - description: ICMP
+        fromPort: -1
+        protocol: icmp
+        toPort: -1
+       # several rules omitted for brevity
+      - description: Service node ports (UDP)
+        fromPort: 30000
+        protocol: udp
+        toPort: 32767
+    vpc:
+      availabilityZoneSelection: Ordered
+      availabilityZoneUsageLimit: 3
+      cidrBlock: 10.0.0.0/16
+  region: us-east-2
+  s3Bucket:
+    name: openshift-bootstrap-data-capi-example-kbdpg
+    presignedURLDuration: 1h0m0s
+status:
+  networkStatus:
+    apiServerElb:
+      attributes: {}
+  ready: false
+```
+
+This manifest describes the network, load balancers, security group rules, etc. that will be used or provisioned for
+the target cluster. Control plane machines are provisioned by [AWSMachines][AWSMachines] & [CAPI Machines][CAPIMachines]
+manifests.
+
+##### Additional Infrastructure
+
+The Installer will generate any additional infrastructure that is needed but not handled by the CAPI provider, either
+because it is out of scope (e.g. IAM, DNS) or not adopted upstream (e.g. split-horizon load balancers). The Installer
```

Comment:
I assume that for any features we think would be in-scope upstream, we will do our best to contribute them there, and we would only use this hook approach if upstream acceptance is not possible. If that's accurate, I think it's worth mentioning.

Reply:
That's a good point, and something we should also document as part of the hooks as well, as a warning or such

Reply:
documented